src/Components/CreateImageWizardV2/steps/Oscap/Oscap.tsx

```diff
@@ -0,0 +1,274 @@
+              <HelpIcon />
+            </Button>
+          </Popover>
+        </>
+      }
+    >
+      <Select
+        ouiaId="profileSelect"
+        variant={SelectVariant.typeahead}
+        onToggle={handleToggle}
+        onSelect={handleSelect}
+        onClear={handleClear}
+        selections={profileName}
+        isOpen={isOpen}
+        placeholderText="Select a profile"
+        typeAheadAriaLabel="Select a profile"
+        isDisabled={!isSuccess}
+        onFilter={(_event, value) => {
+          if (profiles) {
+            return [
+              <OScapNoneOption
+                setProfileName={setProfileName}
+                key="oscap-none-option"
+              />,
+            ].concat(
+              profiles.map((profile_id, index) => {
+                return (
+                  <OScapSelectOption
+                    key={index}
+                    profile_id={profile_id}
+                    setProfileName={setProfileName}
+                    input={value}
+                  />
+                );
+              })
+            );
+          }
+        }}
+      >
+        {options}
+      </Select>
+      {isError && (
+        <Alert
+          title="Error fetching the profiles"
+          variant="danger"
+          isPlain
+          isInline
+        >
+          Cannot get the list of profiles
+        </Alert>
+      )}
+    </FormGroup>
+  );
+};
+
+type OScapNoneOptionPropType = {
+  setProfileName: (name: string) => void;
+};
+
+const OScapNoneOption = ({ setProfileName }: OScapNoneOptionPropType) => {
+  return (
+    <SelectOption
+      value={undefined}
+      onClick={() => {
+        setProfileName('None');
+      }}
+    >
+      <p>{'None'}</p>
+    </SelectOption>
+  );
+};
+
+type OScapSelectOptionPropType = {
+  profile_id: DistributionProfileItem;
+  setProfileName: (name: string) => void;
+  input?: string;
+};
+
+const OScapSelectOption = ({
+  profile_id,
+  setProfileName,
+  input,
+}: OScapSelectOptionPropType) => {
+  const release = useAppSelector((state) => selectDistribution(state));
+  const { data } = useGetOscapCustomizationsQuery({
+    distribution: release,
+    profile: profile_id,
+  });
+
+  if (
+    input &&
+    !data?.openscap?.profile_name?.toLowerCase().includes(input.toLowerCase())
+  ) {
+    return null;
+  }
+
+  return (
+    <SelectOption
+      key={profile_id}
+      value={profile_id}
+      onClick={() => {
+        if (data?.openscap?.profile_name) {
+          setProfileName(data?.openscap?.profile_name);
+        }
+      }}
+    >
+      <p>{data?.openscap?.profile_name}</p>
+    </SelectOption>
+  );
+};
+
+/**
+ * Component to prompt the use with two choices:
+ * - to add a profile, in which case the ProfileSelector will allow the user to
+ *   pick a profile to be stored in the `oscap-profile` variable.
+ * - to not add a profile, in which case the `oscap-profile` form state goes
+ *   undefined.
+ */
```

Comment:
I think this is also no longer true as we changed the structure of the step. The comment needs to get updated or removed.